src/common/decorators/cookie-getter.decorator.ts

```diff
@@ -1,12 +1,22 @@
 import { ExecutionContext, UnauthorizedException, createParamDecorator } from '@nestjs/common';
 import ErrorsTypes from 'src/errors/errors.enum';
 
-export const CookieGetter = createParamDecorator(
+export const CookieOrHeaderGetter = createParamDecorator(
   async (data: string, context: ExecutionContext): Promise<string> => {
     const request = context.switchToHttp().getRequest();
-    const refreshToken = request.cookies[data];
+    let refreshToken = request.cookies[data];
 
-    if (!refreshToken) throw new UnauthorizedException(ErrorsTypes.UNAUTHORIZED_TOKEN_NOT_FOUND);
+    if (!refreshToken) {
+      const refreshHeader = request.headers[data];
+
+      if (!refreshHeader) throw new UnauthorizedException(ErrorsTypes.UNAUTHORIZED_TOKEN_NOT_FOUND);
+
+      const bearer = refreshHeader.split(' ')[0];
+      refreshToken = refreshHeader.split(' ')[1];
```

Comment:
use rest here to not split string twice [bearer, token] = refreshHeader.split(' ')